### DISCUSSION TITLE
Trouble installing Wagmi - multi babel-polyfill ./app/app.js eventsource-polyfill webpack-hot-middleware/client

### DISCUSSION THREAD

**Discussion by straightupjac** [wevm/wagmi#1070] [Q&A]
I am trying to use wagmi with an older code base and having issues with getting it to work:

```ERROR in ./node_modules/wagmi/dist/wagmi.cjs.dev.js 59:15
Module parse failed: Unexpected token (59:15)
You may need an appropriate loader to handle this file type.
|     return Object.entries(cacheRef).reduce((curr, _ref2) => {
|       let [key, value] = _ref2;
>       return { ...curr,
|         [key]: findAndReplace(value, {
|           find,
 @ ./node_modules/wagmi/dist/wagmi.cjs.js 6:19-48
 @ ./app/modules/app/components/root.js
 @ ./app/modules/app/routes.js
 @ ./app/routes/routes.js
 @ ./app/routes/index.js
 @ ./app/app.js
 @ multi babel-polyfill ./app/app.js eventsource-polyfill webpack-hot-middleware/client
```

Has anyone seen an error like this before?

these are the npm and node versions
```
  "engines": {
    "npm": "6.14.4",
    "node": "12.19.1"
  },
```
Unfortunately, updating the node or npm versions won't be possible because of other dependencies in the code base

wagmi and ethers
```
"wagmi": "^0.5.0"
 "ethers": "^5.7.1",
```

**Comment by tmm** [comment]
Looks like you are targeting an older version of JS. wagmi supports es2021 and up by default. For older versions, you will need to update your config for transforming the code.

The specific error seems related to the [spread operator](https://babeljs.io/docs/en/babel-plugin-transform-spread).
